dev/provision_salt.sh

```diff
@@ -19,7 +19,7 @@ function install_salt() {
     sudo apt update
     sudo apt install salt-minion
   elif [[ "$(uname)" == 'Darwin' ]]; then
-    curl https://repo.saltstack.com/osx/salt-3002.1-py3-x86_64.pkg -o /tmp/salt.pkg
```

Comment:
From a security standpoint, is it preferred that we update the URL instead of following redirects?

Reply:
removed the redirect and renamed the URLs to point to their new saltproject domain name.

@godofredoc for future use, whats the security recommendation on redirects?